src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -70,40 +70,194 @@ object SignedTransaction {
       chainId: Option[Byte]
   ): SignedTransaction = {
     val bytes = bytesToSign(tx, chainId)
-    val sig = ECDSASignature.sign(bytes, keyPair, chainId)
-    SignedTransaction(tx, sig)
+    val sig = ECDSASignature.sign(bytes, keyPair)
+    SignedTransaction(tx, getEthereumSignature(tx, sig, chainId))
   }
 
-  private def bytesToSign(tx: Transaction, chainId: Option[Byte]): Array[Byte] =
-    chainId match {
+  private[domain] def bytesToSign(tx: Transaction, chainId: Option[Byte]): Array[Byte] =
+    tx match {
+      case legacyTransaction: LegacyTransaction => getLegacyBytesToSign(legacyTransaction, chainId)
+      case twal: TransactionWithAccessList      => getTWALBytesToSign(twal)
+    }
+
+  private def getLegacyBytesToSign(legacyTransaction: LegacyTransaction, chainIdOpt: Option[Byte]): Array[Byte] =
+    chainIdOpt match {
       case Some(id) =>
-        chainSpecificTransactionBytes(tx, id)
+        chainSpecificTransactionBytes(legacyTransaction, id)
       case None =>
-        generalTransactionBytes(tx)
+        generalTransactionBytes(legacyTransaction)
     }
 
-  def getSender(tx: SignedTransaction): Option[Address] =
-    Option(txSenders.getIfPresent(tx.hash)).orElse(calculateSender(tx))
+  /** Transaction specific piece of code.
+    * This should be moved to the Signer architecture once available.
+    *
+    * Convert a RLP compatible ECDSA Signature to a raw crypto signature.
+    * Depending on the transaction type and the block number, different rules are
+    * used to enhance the v field with additional context for signing purpose and networking
+    * communication.
+    *
+    * Currently, both semantic data are represented by the same data structure.
+    *
+    * @see getEthereumSignature for the reciprocal conversion.
+    * @param signedTransaction the signed transaction from which to extract the raw signature
+    * @return a raw crypto signature, with only 27 or 28 as valid ECDSASignature.v value
+    */
+  private def getRawSignature(
+      signedTransaction: SignedTransaction
+  )(implicit blockchainConfig: BlockchainConfig): ECDSASignature =
+    signedTransaction.tx match {
+      case _: LegacyTransaction =>
+        val chainIdOpt = extractChainId(signedTransaction)
+        getLegacyTransactionRawSignature(signedTransaction.signature, chainIdOpt)
+      case _: TransactionWithAccessList =>
+        getTWALRawSignature(signedTransaction.signature)
+      case _ => throw new IllegalArgumentException(s"Transaction type not supported for $signedTransaction")
```

Comment:
With the current code base, it shouldn't.
New transaction types are expected to be introduced on a regular basis, so this will change.
Currently, Transaction is a sealed trait, so by removing the case _ match, any new transaction not explicitely handled should be caught by the compiler, but since warnings are not generating errors, this won't be visible, until the running client raises a match exception.
The IllegalArgumentException somehow tries to provided a better error.